src/red.ts

```diff
@@ -237,12 +238,22 @@ export const serializedRedEnvSourceText = (function redEnvFactory(blueEnv: Membr
         }
     }
 
-    function lockShadowTarget(shadowTarget: RedShadowTarget, originalTarget: RedProxyTarget) {
-        // copying all own properties into the shadowTarget
-        const targetKeys = ownKeys(originalTarget);
+    function copyBlueDescriptorsIntoShadowTarget(shadowTarget: RedShadowTarget, originalTarget: RedProxyTarget) {
+        const normalizedBlueDescriptors = getOwnPropertyDescriptors(originalTarget);
+        const targetKeys = ownKeys(normalizedBlueDescriptors);
+        const redDescriptors = create(null);
         for (let i = 0, len = targetKeys.length; i < len; i += 1) {
-            copyBlueDescriptorIntoShadowTarget(shadowTarget, originalTarget, targetKeys[i]);
```

Comment:
`copyBlueDescriptorIntoShadowTarget` who is using this now? should this be removed?

Reply:
`copyBlueDescriptorIntoShadowTarget` is still being used in 2 places that cannot switch to `copyBlueDescriptorsIntoShadowTarget`.